debian/check_control

```diff
@@ -33,7 +33,7 @@ LIBNO=$(echo "${VERSION}" | sed -E "s/([0-9]+\.[0-9]+\.[0-9]+).*/\1/;s/\./-/g")
 LIBNAME=$(grep "^Source:" debian/control | sed 's/Source: *//;s/-[0-9].*$//')
 
 # Check version number in control file
-sed -i -e "s/${LIBNAME}-[^ ]*/${LIBNAME}-${LIBNO}/g" debian/control
+sed -i -e "s/${LIBNAME}-[0-9\-]+/${LIBNAME}-${LIBNO}/g" debian/control
```

Comment:
```suggestion
sed -i -e "s/${LIBNAME}-[0-9\-]\+/${LIBNAME}-${LIBNO}/g" debian/control
```

Reply:
I am not sure about your suggested change. I meant a literal minus sign, but not a literal plus sign?

Reply:
Well, and usually I recommend putting the verbatim `-` first in a range, so there is no reading - mistake possible:

```suggestion
sed -i -e "s/${LIBNAME}-[-0-9]\+/${LIBNAME}-${LIBNO}/g" debian/control
```

See https://confluence.desy.de/pages/viewpage.action?spaceKey=DOOCS&title=DOOCS+Lectures+and+Tutorials#DOOCSLecturesandTutorials-Lecture-DOOCSServer-Wednesday,17.11.2021at2pm

![image](https://github.com/user-attachments/assets/23cf97fa-30ea-4d6c-9ceb-750c87227ca0)

Reply:
Ah, that is useful... but why would we escape the plus sign? The shell does not do anything to it, right?

Reply:
Added a version comment to the script and moved the minus sign to the front of the character class. I can still add the backslash in front of the plus if you prefer, but to my eyes it is superfluous (is it?).

Reply:
well, in sed regular mode `+` is a literal `+` but you want a control-character-plus.
In expert  mode (`sed -E`) it is the opposite :grimacing:

Reply:
![image](https://github.com/user-attachments/assets/3f34e8c1-3650-4879-ac4d-d89f70879077)

(sed not mentioned, but you get the idea)

Reply:
> well, in sed regular mode `+` is a literal `+` but you want a control-character-plus. In expert mode (`sed -E`) it is the opposite 😬

Aw. 🤦 Thanks for the Unix coaching. Changed and force-pushed.

Reply:
> add the backslash in front of the plus if you prefer, but to my eyes it is superfluous (is it?).

Well, I tried the script as provided by you and it did not change the numbers in `control`, so I had a look.

Sed 'normal' mode a '+' is a literal, so it would match only if the package name is `gul17-1+` ;-)

If you change `LIBNO` and call `debian/check_control` from the command line it should prepare everything, to try.

Did it work on your machine? Then maybe you have some `sed` preferences set.
Or do I have? No. I just have `grep` always running in 'expert' mode.

![image](https://github.com/user-attachments/assets/143f2539-7bf0-4ee2-9eaf-fb5b2d6e4fba)